sympy/solvers/ode.py

```diff
@@ -4003,6 +4012,47 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+def _check_substitution_type_match(eq, func):
+    r"""
+    Matches any differential equation that `order_reducing_substitution` can solve.
+    For this to work equation should have a minimum order of derivative to be 1, i.e.,
+    `f(x)` should not be present.
+    """
+    fx = func
+    assert len(func.args) == 1  # ODE only handles functions of 1 variable
+    x = func.args[0]
+    D = Dummy()
+    for d in ordered(eq.atoms(Derivative)):
+        try:
+            (v, c), = d.variable_count  # ValueError if not a singleton
+            if v != x:
+                raise ValueError  # not a derivative of interest
+        except ValueError:
+            continue
+        # d was the lowest-ordered derivative of interest
+        if eq.subs(d, D).has(fx):
+            return  # it didn't clear all occurrences of f(x)
+        return {'var': c}
+
+# Use repeated substitution until we do not have a function independent of derivative
+def ode_order_reducing_substitution(eq, func, order, match):
+    r"""
+    Substitutes lowest order derivate in equation to function with order
+    of derivative as 0.Eg `f^(n)(x) = g(x)`, where n is the least order derivate.
+    match[var] or n here is how many times the function if solved is to be integrated
+    to  get f(x) since `g(x) = f^(match[var])(x)`.
+    """
+    x = func.args[0]
+    f = func.func
+    while True:
+        g = Function(Dummy().name)
+        if not eq.has(g(x)):
+            break
+    eq = eq.subs(f(x).diff(x, match['var']), g(x))
+    eq = dsolve(eq)
+    eq = dsolve(eq.subs(g(x), f(x).diff(x, match['var'])))
```

Comment:
You need to tell dsolve explicitly to solve for g(x) as in `dsolve(..., g(x))`.

Reply:
If any g(x) or f(x) is found it wouldn't be able to solve and would raise NotImplemented error. Is this what is required? But no such equation with two function is solvable unless g(x) is explicitly mentioned like sine,cosine etc
which would not be be replaced by substituted.